build/build-composer-json.php

```diff
@@ -142,5 +141,28 @@ function fetchAllData($url, Client $client) {
   }
 
   ksort($composer['conflict']);
-  file_put_contents(__DIR__ . '/' . $target[$core] . '/composer.json', json_encode($composer, JSON_UNESCAPED_SLASHES | JSON_PRETTY_PRINT | JSON_UNESCAPED_UNICODE) . "\n");
+  file_put_contents(__DIR__ . '/' . $target[$core_compat] . '/composer.json', json_encode($composer, JSON_UNESCAPED_SLASHES | JSON_PRETTY_PRINT | JSON_UNESCAPED_UNICODE) . "\n");
+}
+
+/**
+ * @param $result
+ *
+ * @return int
+ */
+function getCoreCompat($result) {
+  switch ($result->field_release_category) {
+    case 'obsolete':
+      $core_compat = -1;
+      break;
+    case 'legacy':
+      $core_compat = 7;
+      break;
+    case 'current':
+      // Drupal's module API goes no higher than 8. Drupal 9 core advisories are published in this project's 8.x branch.
```

Comment:
```suggestion
      // packages.drupal.org/8 is used for 8.x-* versions and all future semantically-versioned releases, which might be compatible with 8 and/or 9 & later.
```

"Core compatibility" is much more complex than 7, 8, or 9 nowadays. I think this really means which packages.drupal.org composer endpoint this corresponds with.

Reply:
Yes exactly. We need a word for that.